plugins/jenkins/src/components/BuildsPage/lib/CITable/CITable.tsx

```diff
@@ -174,48 +173,16 @@ const generatedColumns: TableColumn[] = [
     title: 'Actions',
     sorting: false,
     render: (row: Partial<Project>) => {
-      const ActionWrapper = () => {
-        const [isLoadingRebuild, setIsLoadingRebuild] = useState(false);
-        const { allowed, loading } = useEntityPermission(
-          jenkinsExecutePermission,
-        );
-
-        const alertApi = useApi(alertApiRef);
-
-        const onRebuild = async () => {
-          if (row.onRestartClick) {
-            setIsLoadingRebuild(true);
-            try {
-              await row.onRestartClick();
-              alertApi.post({
-                message: 'Jenkins re-build has successfully executed',
-                severity: 'success',
-              });
-            } catch (e) {
-              alertApi.post({
-                message: `Jenkins re-build has failed. Error: ${e.message}`,
-                severity: 'error',
-              });
-            } finally {
-              setIsLoadingRebuild(false);
-            }
-          }
-        };
-
-        return (
-          <Tooltip title="Rerun build">
-            <>
-              {isLoadingRebuild && <Progress />}
-              {!isLoadingRebuild && (
-                <IconButton onClick={onRebuild} disabled={loading || !allowed}>
-                  <RetryIcon />
-                </IconButton>
-              )}
-            </>
-          </Tooltip>
-        );
-      };
-      return <ActionWrapper />;
+      return row.lastBuild?.url ? (
+        <div style={{ width: '98px' }}>
+          <IconButton href={row.lastBuild.url} target="_blank">
+            <VisibilityIcon />
+          </IconButton>
+          <IconButton href={`${row.lastBuild.url}replay`} target="_blank">
+            <RetryIcon />
+          </IconButton>
+        </div>
+      ) : null;
```

Comment:
I don't understand why you would link to the UI for that.
Replay will work fine with parameters it will just use the same parameters as was used before.

The only reason to go to the replay page would be to modify the build script which is an advanced action which requires elevated permissions users wouldn't normally have.

Reply:
@timja thank you! 👍

@tsoslow, given the above comment I think we need to keep the retry action in place and skip the replay one.

Would it make sense to narrow this down to just add the view link? I'm assuming the replay link would be available from there too.

Reply:
Thanks for the help! The process that got me here is a little convoluted, but it starts because the existing rebuild functionality does not work for us. We always get a "Jenkins re-build has failed. Error: Request failed with 400 Error" popup.

It appears that the problem is that all of our Jenkins workflows require parameters. The jenkins-backend plugin is triggering a new build instead of replaying the existing build because that's the only option that the jenkins sdk provides (also noted in the plugin https://github.com/backstage/backstage/blob/master/plugins/jenkins-backend/src/service/jenkinsApi.ts#L167-L171). And it doesn't appear to have an ability to pass through parameters, nor does that seem like an easy path.

I originally wanted to provide an option to instead display a view link or replay link for teams that had this problem, but I removed the experimental configuration option after @Rugvip let me know that it is not encouraged.

I would love guidance on the direction I should take. Perhaps the issue is not required parameters at all, and I'm doing something wrong and it should just work in its current version? I've had difficulty getting useful information back from the jenkins library and our Jenkins server. All it will return is "jenkins: job.build: bad request".

Thank you again @Rugvip and @timja for your help and patience!

Reply:
I would suggest either adding support for it to https://github.com/silas/node-jenkins
or just skipping the SDK for this, it doesn't do much.

the comment notes the API that needs to be called.